### PR TITLE
[Merged by Bors] - feat: Provide glue between `AddCommGroup` and `Module ℤ`

### PR DESCRIPTION
**Pull request by erdOne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/Torsion.lean [hunk 1]
```diff
@@ -720,6 +720,19 @@ theorem noZeroSMulDivisors_iff_torsion_eq_bot : NoZeroSMulDivisors R M ↔ torsi
             exact ⟨⟨a, mem_nonZeroDivisors_of_ne_zero ha⟩, hax⟩ }
 #align submodule.no_zero_smul_divisors_iff_torsion_eq_bot Submodule.noZeroSMulDivisors_iff_torsion_eq_bot
 
+lemma Submodule.torsion_int {G} [AddCommGroup G] :
+    (Submodule.torsion ℤ G).toAddSubgroup = AddCommGroup.torsion G := by
+  ext x
+  refine ((isOfFinAddOrder_iff_zsmul_eq_zero (x := x)).trans ?_).symm
+  simp [mem_nonZeroDivisors_iff_ne_zero]
+
+lemma AddMonoid.IsTorsionFree_iff_noZeroSMulDivisors {G : Type*} [AddCommGroup G] :
```
Comment (jcommelin) [inline Mathlib/Algebra/Module/Torsion.lean:729]: Shall we try to make both directions instances?

File: Mathlib/LinearAlgebra/FreeModule/PID.lean [hunk 2]
```diff
@@ -422,6 +422,10 @@ theorem Module.free_of_finite_type_torsion_free' [Module.Finite R M] [NoZeroSMul
   exact Module.Free.of_basis b
 #align module.free_of_finite_type_torsion_free' Module.free_of_finite_type_torsion_free'
 
+theorem Module.free_iff_noZeroSMulDivisors [Module.Finite R M] :
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/FreeModule/PID.lean:425]: Idem dito?

File: Mathlib/Algebra/Module/Torsion.lean (hunk 1 above)
Comment (erdOne) [reply Mathlib/Algebra/Module/Torsion.lean:729]: `AddMonoid.IsTorsionFree` is not a class.

File: Mathlib/LinearAlgebra/FreeModule/PID.lean (hunk 2 above)
Comment (erdOne) [reply Mathlib/LinearAlgebra/FreeModule/PID.lean:425]: Maybe. I'll try later (or maybe in #9345)

**Comment by erdOne** [comment]
!bench

File: Mathlib/Algebra/Module/Torsion.lean (hunk 1 above)
Comment (jcommelin) [reply Mathlib/Algebra/Module/Torsion.lean:729]: Fair enough. Maybe it should. But that's not for this PR.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**